File: fsharp/bob/Bob.fs
module Bob

open System.Text.RegularExpressions

let response (input: string): string =
    let yelling = input = input.ToUpper() && Regex.IsMatch(input, @"[a-zA-Z]")
    let silence = input.Trim() = ""
    let question = input.TrimEnd().EndsWith("?")
    
    if silence
    then "Fine. Be that way!"
    else if question
         then if yelling
              then "Calm down, I know what I'm doing!"
              else "Sure."
         else if yelling
              then "Whoa, chill out!"
              else "Whatever."

File: javascript/bob/bob.js
export const hey = message => {
  const silence = message.trim() == "";

  if (silence) {
    return "Fine. Be that way!";
  }

  const yelling = message == message.toUpperCase() && message.match(/[a-z]/i);
  const question = message.trimEnd().endsWith("?");

  if (question && yelling) {
    return "Calm down, I know what I'm doing!";
  } else if (question) {
    return "Sure.";
  } else if (!question && yelling) {
    return "Whoa, chill out!";
  } else {
    return "Whatever.";
  }
};
